docs/source/faq.rst

```diff
@@ -0,0 +1,148 @@
+network protocol or server-side implementation.
+
+Flight SQL is a *wire protocol*.  It specifies the exact commands to
+send to a database to perform various actions like authenticating with
+the database, creating prepared statements, or executing queries.
+Flight SQL specifies the network protocol that the client and the
+server must implement.
+
+One more way of looking at it: an ADBC driver can be written for a
+database purely as a client library.  (That's how the PostgreSQL
+driver in this repository is implemented, for instance—as a wrapper
+around libpq.)  But adding Flight SQL support to a database means
+either modifying the database to run a Flight SQL service, or putting
+the database behind a proxy that translates between Flight SQL and the
+database.
+
+Why not just use Arrow Flight SQL?
+==================================
+
+Continuing on from the previous question:
+
+Because ADBC is client-side, ADBC can support databases that either
+don't support returning Arrow data, or support Arrow data through a
+protocol besides Flight SQL.
+
+Then do we even need Arrow Flight SQL?
+======================================
+
+Continuing on from the previous two questions:
+
+Flight SQL is a concrete protocol that database vendors can implement,
+instead of designing their own protocol.  And Flight SQL also has JDBC
+and ODBC drivers for maximal compatibility.
+
+As an analogy: many databases implement the PostgreSQL wire protocol,
+so that they can gain access to all the PostgreSQL clients, including
+JDBC and ODBC drivers.  (And JDBC/ODBC users can still use other
+drivers to work with other databases.)
+
+For the Arrow ecosystem, we hope databases will implement the Flight
+SQL wire protocol, giving them access to all the Flight SQL clients,
+including ADBC, JDBC, and ODBC drivers.  (And ADBC users can still use
+other drivers to work with other databases.)
+
+So what is the "ADBC Flight SQL driver" then?
+=============================================
+
+The ADBC Flight SQL driver implements the ADBC API standard (which an
+application interacts with) using the Flight SQL wire protocol (which
+a database server exposes).  So it's a generic driver that can talk to
+many databases, and it also implements a generic API (ADBC) designed
+to abstract over multiple databases.
+
+This is a little unusual, in that most database drivers and database
+protocols you'll find were meant for a specific database.  But Flight
+SQL was designed to be agnostic to the database from the start, and so
+was ADBC.  It sounds like they overlap, but they complement each other
+because they operate at different levels of abstraction.  Flight SQL
+targets database server developers, and gives them one protocol they
+can implement in order to reach ADBC, JDBC, and ODBC users.  ADBC
+targets database users, and gives them one Arrow-native API they can
+use to work with both Arrow-native and non-Arrow-native databases.
```

Comment:
It might be worthwhile to explicitly state: Database server developers can provide a Flight SQL interface and they won't have to build and distribute their own drivers for any of ADBC, JDBC, or ODBC. Consumers would be able to leverage the existing Flight SQL drivers for all three instead. Thereby reducing the maintenance burden on database developers.

At least in some form or wording, but I think it's worthwhile to be explicit about that rather than letting people draw that conclusion.